src/psyclone/domain/nemo/transformations/nemo_loop_fuse.py

```diff
@@ -70,27 +71,24 @@ def validate(self, node1, node2, options=None):
         sym_maths = SymbolicMaths.get()
 
         if not sym_maths.equal(node1.start_expr, node2.start_expr):
-            raise TransformationError("Lower loop bounds must be identical, "
-                                      "but are '{0}'' and '{1}'"
-                                      .format(node1.start_expr,
-                                              node2.start_expr))
+            raise TransformationError(f"Lower loop bounds must be identical, "
+                                      f"but are '{node1.start_expr}' and "
+                                      f"'{node2.start_expr}'")
         if not sym_maths.equal(node1.stop_expr, node2.stop_expr):
-            raise TransformationError("Upper loop bounds must be identical, "
-                                      "but are '{0}'' and '{1}'"
-                                      .format(node1.stop_expr,
-                                              node2.stop_expr))
+            raise TransformationError(f"Upper loop bounds must be identical, "
+                                      f"but are '{node1.stop_expr}' and "
+                                      f"'{node2.stop_expr}'")
         if not sym_maths.equal(node1.step_expr, node2.step_expr):
-            raise TransformationError("Step size in loops must be identical, "
-                                      "but are '{0}'' and '{1}'"
-                                      .format(node1.step_expr,
-                                              node2.step_expr))
+            raise TransformationError(f"Step size in loops must be identical, "
+                                      f"but are '{node1.step_expr}' and "
+                                      f"'{node2.step_expr}'")
         loop_var1 = node1.variable
         loop_var2 = node2.variable
 
         if loop_var1 != loop_var2:
-            raise TransformationError("Loop variables must be the same, "
-                                      "but are '{0}' and '{1}'".
-                                      format(loop_var1.name, loop_var2.name))
+            raise TransformationError(f"Loop variables must be the same, "
+                                      f"but are '{loop_var1.name}' and "
+                                      f"'{loop_var2.name}'")
         vars1 = VariablesAccessInfo(node1)
```

Comment:
Maybe this is more generic than this transformation and could be mostly encapsulated in a new DependencyTool entry point (maybe `are_loops_fusable(loop1, loop2)` or `are_loops_independent(loop1, loop2)` ). It does feel this code belogn more on that domain and it would keep `partition` as private method. What do you think?